tools/kicad/kicad_rs/src/parser.rs

```diff
@@ -363,6 +415,22 @@ impl<T: AsRef<str>> SplitCharN for Option<T> {
     }
 }
 
+trait OrDefault<'a, T> {
+    fn or_default(self, default: T) -> T;
+}
+
+impl<'a, T: Default + PartialEq> OrDefault<'a, T> for T {
+    fn or_default(self, default: T) -> T {
+        if self == Default::default() {
```

Comment:
What does this check do? Where is the default value defined and for what type?

Reply:
Yeah some documentation on trait would be great :+1:

Reply:
Documented in https://github.com/racklet/electronics-prototyping/pull/18/commits/02921b01f1086963a98f6fe3529e84da34d18cc4. [`Default`](https://doc.rust-lang.org/std/default/trait.Default.html) is a built-in trait that enables fetching the type-specific "default" value, e.g. "" (empty string) for string-like types and 0 for number-like types. I'm using it here to make this generic without having to write an `impl` for each type comparing against its respective "zero value".